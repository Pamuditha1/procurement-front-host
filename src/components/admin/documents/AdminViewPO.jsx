import React, {
  useState,
  useEffect,
} from "react";
import { Table } from "reactstrap";
import { Link } from "react-router-dom";
import Loader from "react-loader-spinner";

import getPOs from "../../../services/getPOs";

function AdminViewPO() {
  const [pos, setpos] = useState([]);

  useEffect(() => {
    async function fetchMSRs() {
      const results = await getPOs();
      setpos(results);
    }

    fetchMSRs();
  }, []);

  return (
    <div data-testid="adminviewpo">
      <h2 data-testid="adminpo-heading">Admin PO</h2>
      {pos?.length === 0 ? (
        <div
          className="container text-center"
          style={{ width: "793px" }}
        >
          <Loader
            type="Puff"
            color="#050A30"
            height={100}
            width={100}
            timeout={5000}
          />
        </div>
      ) : (
        <div>
          <h6
            className="pl-5 pt-1 pb-1 mb-5 mt-4"
            style={{ backgroundColor: "gray" }}
          >
            PO List
            <Link
              to="/admin/documents"
              style={{ marginLeft: "80%" }}
            >
              <button className="btn btn-outline-light">
                Back
              </button>
            </Link>
          </h6>

          <Table hover borderless>
            <thead className="text-center">
              <tr>
                <th>View</th>
                <th>Status</th>
                <th>PO No</th>
                <th>Date</th>
                <th>Time</th>
                <th>Created</th>
                <th>Confirmed/ Rejected</th>
                <th></th>
              </tr>
            </thead>
            <tbody>
              {pos?.map((p) => {
                return (
                  <tr
                    style={{
                      textAlign: "center",
                    }}
                    key={p._id}
                  >
                    <td>
                      <Link
                        to={`/admin/documents/po/${p._id}`}
                      >
                        <button className="btn btn-primary">
                          View
                        </button>
                      </Link>
                    </td>
                    <td
                      className={
                        p.status === "Confirmed"
                          ? "text-success text-center"
                          : p.status ===
                            "Rejected"
                          ? "text-danger text-center"
                          : p.status ===
                              "Recreated" &&
                            "text-primary text-center"
                      }
                    >
                      {p.status}
                    </td>
                    <td className="text-center">
                      {p.poNo}
                    </td>
                    <td className="text-center">
                      {new Date(
                        p.timeStamp
                      ).toLocaleDateString()}
                    </td>
                    <td className="text-center">
                      {new Date(
                        p.timeStamp
                      ).toLocaleTimeString()}
                    </td>
                    <td className="text-center">
                      {p.createdBy.username}
                    </td>
                    <td className="text-center">
                      {p.approvedBy
                        ? p.approvedBy.username
                        : "Pending..."}
                    </td>
                  </tr>
                );
              })}
            </tbody>
          </Table>
        </div>
      )}
    </div>
  );
}

export default AdminViewPO;
