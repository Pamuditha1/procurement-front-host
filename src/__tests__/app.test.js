import { render, screen, cleanup } from "@testing-library/react";
import * as renderer from "react-test-renderer";
import AdminViewPO from "../components/admin/documents/AdminViewPO";

describe("AdminViewPO Compenent", () => {
    afterEach(() => {
        cleanup();
    });

    test("Should render AdminViewPO component", () => {
        render(<AdminViewPO />);
        const element = screen.getByTestId("adminviewpo");
        expect(element).toBeInTheDocument();
    });

    test("Should matches DOM Snapshot", () => {
        const domTree = renderer.create(<AdminViewPO />).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    test("Should render Admin PO heading", () => {
        render(<AdminViewPO />);
        const element = screen.getByTestId("adminpo-heading");
        expect(element).toHaveTextContent("Admin PO");
    });

    test("test", () => {
        expect(true).toBe(true);
    })
})