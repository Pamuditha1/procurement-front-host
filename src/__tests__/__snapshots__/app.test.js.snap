// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminViewPO Compenent Matches DOM Snapshot 1`] = `
<div
  data-testid="adminviewpo"
>
  <h2
    data-testid="adminpo-heading"
  >
    Admin PO
  </h2>
  <div
    className="container text-center"
    style={
      Object {
        "width": "793px",
      }
    }
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        height={100}
        stroke="#050A30"
        viewBox="0 0 44 44"
        width={100}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="2"
        >
          <circle
            cx="22"
            cy="22"
            r={1}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
          <circle
            cx="22"
            cy="22"
            r={1}
          >
            <animate
              attributeName="r"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`AdminViewPO Compenent Should matches DOM Snapshot 1`] = `
<div
  data-testid="adminviewpo"
>
  <h2
    data-testid="adminpo-heading"
  >
    Admin PO
  </h2>
  <div
    className="container text-center"
    style={
      Object {
        "width": "793px",
      }
    }
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        height={100}
        stroke="#050A30"
        viewBox="0 0 44 44"
        width={100}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="2"
        >
          <circle
            cx="22"
            cy="22"
            r={1}
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
          <circle
            cx="22"
            cy="22"
            r={1}
          >
            <animate
              attributeName="r"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
